eva-server/src/main/java/uk/ac/ebi/eva/server/ws/VariantWSServer.java

```diff
@@ -47,38 +54,65 @@
 @Api(tags = {"variants"})
 public class VariantWSServer extends EvaWSServer {
 
+    @Autowired
+    private VariantEntityRepository variantEntityRepository;
+
+    protected static Logger logger = LoggerFactory.getLogger(FeatureWSServer.class);
+
     @RequestMapping(value = "/{variantId}/info", method = RequestMethod.GET)
 //    @ApiOperation(httpMethod = "GET", value = "Retrieves the information about a variant", response = QueryResponse.class)
     public QueryResponse getVariantById(@PathVariable("variantId") String variantId,
                                         @RequestParam(name = "studies", required = false) List<String> studies,
-                                        @RequestParam("species") String species,
+                                        @RequestParam(name = "species", required = true) String species,
+                                        @RequestParam(name = "annot-ct", required = false) List<String> consequenceType,
+                                        @RequestParam(name = "maf", defaultValue = "", required = false) String maf,
+                                        @RequestParam(name = "polyphen", defaultValue = "", required = false)
+                                                    String polyphenScore,
+                                        @RequestParam(name = "sift", defaultValue = "",required = false)
+                                                    String siftScore,
                                         HttpServletResponse response)
             throws IllegalOpenCGACredentialsException, UnknownHostException, IOException {
         initializeQueryOptions();
 
-        VariantDBAdaptor variantMongoDbAdaptor = DBAdaptorConnector.getVariantDBAdaptor(species);
 
-        if (studies != null && !studies.isEmpty()) {
-            queryOptions.put("studies", studies);
+        if (species == null || species.isEmpty()) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            return setQueryResponse("Please specify a species");
         }
 
-        if (!variantId.contains(":")) { // Query by accession id
-            return setQueryResponse(variantMongoDbAdaptor.getVariantById(variantId, queryOptions));
-        } else { // Query by chr:pos:ref:alt
```

Comment:
This branch of the condition also needs to be supported for the Variant View to work, which means another method will be necessary in the repository.

Even if the all code called it `getAllVariantsByRegion`, this is not a query by region and could probably be generated automatically by Spring Data.